services/requester/requester.go

```diff
@@ -373,16 +372,15 @@ func (e *EVM) EstimateGas(
 	if err != nil {
 		return 0, fmt.Errorf("failed to decode EVM result from gas estimation: %w", err)
 	}
+
 	if evmResult.ErrorCode != 0 {
+		if evmResult.ErrorCode == evmTypes.ExecutionErrCodeExecutionReverted {
+			return 0, errs.NewRevertError(evmResult.ReturnedData)
+		}
 		return 0, getErrorForCode(evmResult.ErrorCode)
 	}
 
-	// This minimum gas availability is needed for:
-	// https://github.com/onflow/go-ethereum/blob/master/core/vm/operations_acl.go#L29-L32
-	// Note that this is not actually consumed in the end.
-	// TODO: Consider moving this to `EVM.dryRun`, if we want the
-	// fix to also apply for the EVM API, on Cadence side.
-	gasConsumed := evmResult.GasConsumed + params.SstoreSentryGasEIP2200 + 1
```

Comment:
so we removed this because we handle it in the EVM core now?

Reply:
Yes, exactly, this is done by `EVM.dryRun` now.